src/TLuaEngine.cpp

```diff
@@ -921,7 +1002,17 @@ std::shared_ptr<TLuaResult> TLuaEngine::StateThreadData::EnqueueFunctionCall(con
     Result->StateId = mStateId;
     Result->Function = FunctionName;
     std::unique_lock Lock(mStateFunctionQueueMutex);
-    mStateFunctionQueue.push_back({ FunctionName, Result, Args, "" });
+    mStateFunctionQueue.push_back({ FunctionName, Result, Args, "", NULL });
+    mStateFunctionQueueCond.notify_all();
+    return Result;
+}
+
+std::shared_ptr<TLuaResult> TLuaEngine::StateThreadData::EnqueueFunctionCall(std::shared_ptr<sol::reference> FunctionRef, const std::vector<TLuaArgTypes>& Args) {
+    auto Result = std::make_shared<TLuaResult>();
+    Result->StateId = mStateId;
+    Result->Function = "anonymous";
```

Comment:
Could we instead use `fmt::format("[[function {:p}]]", FunctionRef->pointer())` or similar? So that people who print() a whole table with functions can find out which function this was, if they want! :)

I didn't know sol::reference, that's kinda exactly what I needed for a lot of cases! Thanks for showing me :)